src/dwn.ts

```diff
@@ -29,10 +36,17 @@ export class DWN {
     // throws exception if message is invalid
     validateMessage(message);
 
-    await method(message, this.DIDResolver);
+    await method(message, this.DIDResolver, this.messageStore);
   }
 };
 
 export type Config = {
   DIDMethodResolvers: DIDMethodResolver[],
+  messageStore: MessageStore
+};
+
+const defaultConfig: Config = {
```

Comment:
I'm curious about your reasonings behind avoiding config defaults. Is it so that all developers must familiarize themselves with potential config options? I tend to favor default config quite a bit to reduce cognitive overhead for devs. I can't think of a single service that i've used that didn't set defaults that could be overridden

Reply:
I think for libraries/CLI, defaults make sense, mainly 'coz there could be so many. But for a service, not as much. The reasoning is ironically similar to the cognitive overhead you mentioned:

1. Yes to your point about knowing the config options, often I don't know about a service config parameter 'coz it's optional.
1. As much as possible I'd like the dev/devOps to be able to look at one config file that would contain all the parameters used to instantiate the service as opposed to: a config file containing a subset, then having to search for "default values" in the code, maybe in multiple files, when debugging and getting confused as to where a value came from.
1. Not having defaults also forces us to be more disciplined about what we expose to the dev/devops. As in, if a default value can suffice most of the time, I'd consider just remove it entirely so the devs really don't have to know about it. So when a param does get exposed, the dev/devop really ought to know of its existence and set a meaningful value according to their services' need.

This is not to say that we can't have default/suggested values external to the code in say json config files, at least that'll be a single place where they know how their service is configured.

I know this is entirely subjective, so no right or wrong really, seems you are also opinionated about this one, so happy to hear what you have to say after seeing my reasoning. And if you still want to keep them optional, so be it, but I do want to know if you'll be wanting for contributors/me to make parameters optional wherever possible? As I said, I'd prefer to have no parameter wherever possible. In my mind I would certainly would avoid any optional parameters for all internal facing classes such as `BlockstoreLevel`.

Reply:
👍🏽